From 0a6d945f0dc1b1941aa4265ce68718582f5bea32 Mon Sep 17 00:00:00 2001
From: Christoph Reiter <reiter.christoph@gmail.com>
Date: Fri, 2 Jul 2021 19:11:46 +0200
Subject: [PATCH 105/N] [squash] CI: Clean up

this is all no longer needed now

Needs to be merged into the CI setup
---
 .github/workflows/mingw.yml | 45 +++++--------------------------------
 1 file changed, 6 insertions(+), 39 deletions(-)

diff --git a/.github/workflows/mingw.yml b/.github/workflows/mingw.yml
index 8486bb4..d45a9c2 100644
--- a/.github/workflows/mingw.yml
+++ b/.github/workflows/mingw.yml
@@ -50,8 +50,7 @@ jobs:
         shell: msys2 {0}
         run: |
           set -ex
-          pkgver='3.9.5'
-          _pybasever='3.9'
+
           CFLAGS="-O2"
           
           PREFIX_WIN=$(cygpath -wm ${MINGW_PREFIX})
@@ -67,35 +66,6 @@ jobs:
           CFLAGS+=" -DNDEBUG "
           CXXFLAGS+=" -DNDEBUG "
           _extra_config+=("--enable-optimizations")
-          # Fix MS_DLL_ID undeclared in PC/dl_nt.c
-          f3f=0
-          re='^[0-9]+$'
-          part3=${pkgver##*.}
-          if ! [[ $part3 =~ $re ]]; then
-              c=
-              while test -n "$part3"; do
-                  b=${part3:0:1}
-                  if ! [[ $b =~ $re ]]; then
-                      break
-                  fi
-                  c+=${b}
-                  part3=${part3:1}
-              done
-              f3f=$((f3f + 1000 * $((${c}))))
-              part3Len=${#part3}
-              if [ "$part3Len" -ge "1" ]; then
-                  f3f=$((f3f + 10 * $((0x${part3:0:1}))))
-                  if [ "$part3Len" -ge "2" ]; then
-                      f3f=$((f3f + ${part3:1:1}))
-                  fi
-              fi
-          else
-              f3f=$((f3f + 1000 * ${part3}))
-          fi
-          CFLAGS+=" -DMS_DLL_ID=\\\"${_pybasever}\\\""
-          CFLAGS+=" -DFIELD3=${f3f}"
-          CFLAGS+=" -DORIGINAL_FILENAME=\\\"libpython${_pybasever}.dll\\\""
-          rc_flags=" -DMS_DLL_ID=\\\\\\\"${_pybasever}\\\\\\\" -DFIELD3=${f3f} -DORIGINAL_FILENAME=\\\\\\\"libpython${_pybasever}.dll\\\\\\\""
 
           # Workaround for conftest error on 64-bit builds
           export ac_cv_working_tzset=no
@@ -131,7 +101,6 @@ jobs:
             --without-c-locale-coercion \
             --enable-loadable-sqlite-extensions \
             "${_extra_config[@]}" \
-            RCFLAGS="$rc_flags" \
             OPT=""
           # We patch importlib which is embedded in C headers, so regenerate them
           make regen-importlib
@@ -147,9 +116,8 @@ jobs:
         shell: msys2 {0}
         run: |
           set -ex
-          
-          pkgver='3.9.5'
-          _pybasever='3.9'
+
+          _pybasever=$(./python.exe -c "import sys; print(sys.winver);")
           srcdir=.
 
           # now install things
@@ -160,7 +128,6 @@ jobs:
           MSYSTEM=MINGW \
           MSYS2_ARG_CONV_EXCL="--prefix=;--install-scripts=;--install-platlib=" \
           make -j1 install DESTDIR="${pkgdir}"
-          VERABI=${_pybasever}
 
           # gdb pretty printers for debugging Python itself; to use:
           # python
@@ -169,10 +136,10 @@ jobs:
           # reload(python_gdb)
           # end
 
-          cp -f "${pkgdir}${MINGW_PREFIX}"/lib/python${_pybasever}/config-${VERABI}/libpython${VERABI}.dll.a "${pkgdir}${MINGW_PREFIX}"/lib/libpython${VERABI}.dll.a
+          cp -f "${pkgdir}${MINGW_PREFIX}"/lib/python${_pybasever}/config-${_pybasever}/libpython${_pybasever}.dll.a "${pkgdir}${MINGW_PREFIX}"/lib/libpython${_pybasever}.dll.a
 
           # Need for building boost python module
-          cp -f "${pkgdir}${MINGW_PREFIX}"/lib/python${_pybasever}/config-${VERABI}/libpython${VERABI}.dll.a "${pkgdir}${MINGW_PREFIX}"/lib/libpython${_pybasever}.dll.a
+          cp -f "${pkgdir}${MINGW_PREFIX}"/lib/python${_pybasever}/config-${_pybasever}/libpython${_pybasever}.dll.a "${pkgdir}${MINGW_PREFIX}"/lib/libpython${_pybasever}.dll.a
 
           for fscripts in 2to3 2to3-${_pybasever} idle3 idle${_pybasever} pydoc3 pydoc${_pybasever}; do
               sed -i "s|$(cygpath -w ${MINGW_PREFIX} | sed 's|\\|\\\\|g')/bin/python${_pybasever}.exe|/usr/bin/env python${_pybasever}.exe|g" "${pkgdir}${MINGW_PREFIX}"/bin/${fscripts}
@@ -185,7 +152,7 @@ jobs:
           cp "${pkgdir}${MINGW_PREFIX}"/bin/idle3 "${pkgdir}${MINGW_PREFIX}"/bin/idle
           cp "${pkgdir}${MINGW_PREFIX}"/bin/pydoc3 "${pkgdir}${MINGW_PREFIX}"/bin/pydoc
 
-          sed -i "s|#!${pkgdir}${MINGW_PREFIX}/bin/python${VERABI}.exe|#!/usr/bin/env python${_pybasever}.exe|" "${pkgdir}${MINGW_PREFIX}"/lib/python${_pybasever}/config-${VERABI}/python-config.py
+          sed -i "s|#!${pkgdir}${MINGW_PREFIX}/bin/python${_pybasever}.exe|#!/usr/bin/env python${_pybasever}.exe|" "${pkgdir}${MINGW_PREFIX}"/lib/python${_pybasever}/config-${_pybasever}/python-config.py
 
           # fix permissons
           find ${pkgdir}${MINGW_PREFIX} -type f \( -name "*.dll" -o -name "*.exe" \) | xargs chmod 0755
-- 
2.32.0

